doc/source/ray-air/check-ingest.rst

```diff
@@ -0,0 +1,241 @@
+user-defined function to preprocess batches of data, and (3) runs an AIR Trainer with the given dataset and preprocessor.
+
+.. figure:: images/ingest.svg
+
+..
+  https://docs.google.com/drawings/d/1FPta-TvaL7TQoSRp5ofts0dru6mw1Xes-35m5XuqCBc/edit
+
+Let's walk through the stages of what happens when ``Trainer.fit()`` is called.
+
+**Read**: First, AIR will read the Dataset into the Ray object store by calling ``ds.fully_executed()`` on the datasets
+that you pass to the Trainer. Dataset blocks that don't fit into memory will be spilled to disk. Note that when you create
+the dataset initially, typically only the first block and block metadata is read into memory. The rest of the blocks are
+not loaded until ``fit`` is called.
+
+**Preprocessing**: Next, if a preprocessor is defined, AIR will ``fit`` the preprocessor (e.g., for stateful preprocessors) on the
+``"train"`` dataset, and then ``transform`` all given datasets with the fitted preprocessor. This is done by calling ``prep.fit_transform()``
+on the train dataset passed to the Trainer, followed by ``prep.transform()`` on remaining datasets. Preprocessors use Dataset APIs to execute
+preprocessing in a parallelized way across the cluster. Both read and preprocessing stages use Ray tasks under the hood.
+
+**Training**: Finally, AIR passes a reference to the preprocessed dataset to Train workers (Ray actors) launched by the Trainer. Each worker then
+typically calls ``iter_batches``, ``to_tf``, or ``to_torch`` to iterate over the dataset reference retrieved by ``get_dataset_shard``.
+These read methods load blocks of the dataset into the local worker's memory in a streaming fashion, only fetching / prefetching a
+limited number of blocks at once. Workers loop over the dataset blocks repeatedly until training completes.
+
+Bulk Ingest
+~~~~~~~~~~~
+
+By default, AIR loads all Dataset blocks into the object store at the start of training. This provides the best performance if the
+cluster has enough aggregate memory to fit all the data blocks in object store memory. Note that data often requires more space
+when loaded uncompressed in memory than when resident in storage.
+
+If there is insufficient object store memory, blocks may be spilled to disk during reads or preprocessing. Ray will print log messages
+if spilling is occuring, and you can check this as well with the ``ray memory --stats-only`` utility. If spilling is happening, take
+care to ensure the cluster has enough disk space to handle the spilled blocks. Alternatively, consider using machine with more memory /
+more machines to avoid spilling.
+
+Streamed Ingest
+~~~~~~~~~~~~~~~
+
+This section is a placeholder.
+
+AIR will support streamed ingest and enable it by default by Beta. Streamed ingest is preferable when you are using large
+datasets that don't fit into memory, and also don't need advanced training quality features such as global random shuffle.
```

Comment:
Another important use case for streaming ingest is if we have expensive preprocessors so we only want to transform one window at a time rather than the entire dataset up front (even if the original dataset may fit into memory), right?

Reply:
Yep! I'll make sure to add this when this is implemented.